[rocm-gdb] Unable to build, missing symbol amd_dbgapi_initialize in rocm-dbgapi

Getting the following error when _make_'ing rocm-gdb:

```
[...]
checking amd-dbgapi.h usability... yes
checking amd-dbgapi.h presence... yes
checking for amd-dbgapi.h... yes
checking for amd_dbgapi_initialize in -lrocm-dbgapi... no
configure: error: amdgcn*-*-* requires rocm-dbgapi to be installed, or use --with-rocm-dbgapi=PATH
```

The `rocm-dbgapi` package is build/installed under `/opt/rocm` prefix using `yay -S rocm-dbgapi` and as far as I can tell contains the requested symbol:

```
$ readelf -Ws /opt/rocm/lib/librocm-dbgapi.so | grep amd_dbgapi_initialize
   172: 0000000000026f70   248 FUNC    GLOBAL DEFAULT   10 amd_dbgapi_initialize@@AMD_DBGAPI_0.24
```

Since `amd-dbgapi.h` is found I'm sure its also using the correct library lookup path, but even when using `--with-rocm-dbgapi=/opt/rocm`  or `LDFLAGS="-L/opt/rocm/lib $LDFLAGS"` it aborts during _make_ with the same message...

Some further debugging. Extracting the relevant configure fragment:

```
char amd_dbgapi_initialize ();
int main () {
  return amd_dbgapi_initialize ();
}
```

and building it

```
$ gcc -Wall -g -c test.c
$ gcc -o test test.o -lrocm-dbgapi -L/opt/rocm/lib
/bin/ld: warning: libamd_comgr.so.1, needed by /opt/rocm/lib/librocm-dbgapi.so, not found (try using -rpath or -rpath-link)
/bin/ld: /opt/rocm/lib/librocm-dbgapi.so: undefined reference to `amd_comgr_create_disassembly_info@amd_comgr_1.8'
/bin/ld: /opt/rocm/lib/librocm-dbgapi.so: undefined reference to `amd_comgr_disassemble_instruction@amd_comgr_1.8'
/bin/ld: /opt/rocm/lib/librocm-dbgapi.so: undefined reference to `amd_comgr_destroy_disassembly_info@amd_comgr_1.8'
```

So `rocm-dbgapi` cannot find `amd_comgr`, probably because `/opt/rocm/lib`, so I should be able to get this working with
`-Wl,--rpath=/opt/rocm/lib/` (or `LD_LIBRARY_PATH=/opt/rocm/lib/`).

It is building now, but fails after some time with:

```
  CXX    rocm-tdep.o
../../gdb/rocm-tdep.c: In function 'void info_agents_command(const char*, int)':
../../gdb/rocm-tdep.c:1848:32: error: 'AMD_DBGAPI_AGENT_INFO_SHADER_ENGINE_COUNT' was not declared in this scope; did you mean 'AMD_DBGAPI_AGENT_INFO_EXECUTION_UNIT_COUNT'?
 1848 |       UIOUT_FIELD_INT (num_se, AMD_DBGAPI_AGENT_INFO_SHADER_ENGINE_COUNT);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../gdb/rocm-tdep.c:1836:71: note: in definition of macro 'UIOUT_FIELD_INT'
 1836 |   if ((status = amd_dbgapi_agent_get_info (process_id, agent_list[i], query,  \
      |                                                                       ^~~~~
../../gdb/rocm-tdep.c:1849:32: error: 'AMD_DBGAPI_AGENT_INFO_COMPUTE_UNIT_COUNT' was not declared in this scope; did you mean 'AMD_DBGAPI_AGENT_INFO_EXECUTION_UNIT_COUNT'?
 1849 |       UIOUT_FIELD_INT (num_cu, AMD_DBGAPI_AGENT_INFO_COMPUTE_UNIT_COUNT);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../gdb/rocm-tdep.c:1836:71: note: in definition of macro 'UIOUT_FIELD_INT'
 1836 |   if ((status = amd_dbgapi_agent_get_info (process_id, agent_list[i], query,  \
      |                                                                       ^~~~~
../../gdb/rocm-tdep.c:1850:30: error: 'AMD_DBGAPI_AGENT_INFO_NUM_SIMD_PER_COMPUTE_UNIT' was not declared in this scope; did you mean 'AMD_DBGAPI_AGENT_INFO_MAX_WAVES_PER_EXECUTION_UNIT'?
 1850 |       UIOUT_FIELD_INT (simd, AMD_DBGAPI_AGENT_INFO_NUM_SIMD_PER_COMPUTE_UNIT);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../gdb/rocm-tdep.c:1836:71: note: in definition of macro 'UIOUT_FIELD_INT'
 1836 |   if ((status = amd_dbgapi_agent_get_info (process_id, agent_list[i], query,  \
      |                                                                       ^~~~~
../../gdb/rocm-tdep.c:1851:31: error: 'AMD_DBGAPI_AGENT_INFO_MAX_WAVES_PER_SIMD' was not declared in this scope; did you mean 'AMD_DBGAPI_AGENT_INFO_PCI_VENDOR_ID'?
 1851 |       UIOUT_FIELD_INT (waves, AMD_DBGAPI_AGENT_INFO_MAX_WAVES_PER_SIMD);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../gdb/rocm-tdep.c:1836:71: note: in definition of macro 'UIOUT_FIELD_INT'
 1836 |   if ((status = amd_dbgapi_agent_get_info (process_id, agent_list[i], query,  \
      |                                                                       ^~~~~
make[2]: *** [Makefile:1663: rocm-tdep.o] Error 1
make[2]: Leaving directory '/home/sirow/.cache/yay/rocm-gdb/src/ROCgdb-rocm-3.8.0/build/gdb'
```

--- a/rocm-gdb/.SRCINFO
+++ b/rocm-gdb/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = rocm-gdb
 	pkgdesc = ROCm source-level debugger for Linux, based on GDB
-	pkgver = 3.8.0
+	pkgver = 3.9.0
 	pkgrel = 1
 	url = https://github.com/ROCm-Developer-Tools/ROCgdb
 	arch = x86_64
@@ -16,8 +16,8 @@ pkgbase = rocm-gdb
 	depends = mpfr
 	depends = source-highlight
 	depends = babeltrace
-	source = rocm-gdb-3.8.0.tar.gz::https://github.com/ROCm-Developer-Tools/ROCgdb/archive/rocm-3.8.0.tar.gz
-	sha256sums = a7c11dc30c952587c616bf7769bad603c3bf80522afc8b73ccda5b78d27bed41
+	source = rocm-gdb-3.9.0.tar.gz::https://github.com/ROCm-Developer-Tools/ROCgdb/archive/rocm-3.9.0.tar.gz
+	sha256sums = 0765c96439c0efa145418d210d865b9faed463466d7522274959cc4476a37097
 
 pkgname = rocm-gdb
 

--- a/rocm-gdb/PKGBUILD
+++ b/rocm-gdb/PKGBUILD
@@ -1,6 +1,6 @@
 # Maintainer Torsten Keßler <t dot kessler at posteo dot de>
 pkgname=rocm-gdb
-pkgver=3.8.0
+pkgver=3.9.0
 pkgrel=1
 pkgdesc='ROCm source-level debugger for Linux, based on GDB'
 arch=('x86_64')
@@ -9,7 +9,7 @@ license=('GPL')
 depends=('rocm-dbgapi' 'python' 'guile2.0' 'ncurses' 'expat' 'xz' 'zlib' 'mpfr' 'source-highlight' 'babeltrace')
 makedepends=('texinfo')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/rocm-$pkgver.tar.gz")
-sha256sums=('a7c11dc30c952587c616bf7769bad603c3bf80522afc8b73ccda5b78d27bed41')
+sha256sums=('0765c96439c0efa145418d210d865b9faed463466d7522274959cc4476a37097')
 _dirname="$(basename "$url")-$(basename "${source[0]}" ".tar.gz")"
 
 prepare() {

